exporter/datadogexporter/translate_traces.go

```diff
@@ -0,0 +1,517 @@
+	trace.StatusCodeInvalidArgument:    {message: "INVALID_ARGUMENT", status: http.StatusBadRequest},
+	trace.StatusCodeDeadlineExceeded:   {message: "DEADLINE_EXCEEDED", status: http.StatusGatewayTimeout},
+	trace.StatusCodeNotFound:           {message: "NOT_FOUND", status: http.StatusNotFound},
+	trace.StatusCodeAlreadyExists:      {message: "ALREADY_EXISTS", status: http.StatusConflict},
+	trace.StatusCodePermissionDenied:   {message: "PERMISSION_DENIED", status: http.StatusForbidden},
+	trace.StatusCodeResourceExhausted:  {message: "RESOURCE_EXHAUSTED", status: http.StatusTooManyRequests},
+	trace.StatusCodeFailedPrecondition: {message: "FAILED_PRECONDITION", status: http.StatusBadRequest},
+	trace.StatusCodeAborted:            {message: "ABORTED", status: http.StatusConflict},
+	trace.StatusCodeOutOfRange:         {message: "OUT_OF_RANGE", status: http.StatusBadRequest},
+	trace.StatusCodeUnimplemented:      {message: "UNIMPLEMENTED", status: http.StatusNotImplemented},
+	trace.StatusCodeInternal:           {message: "INTERNAL", status: http.StatusInternalServerError},
+	trace.StatusCodeUnavailable:        {message: "UNAVAILABLE", status: http.StatusServiceUnavailable},
+	trace.StatusCodeDataLoss:           {message: "DATA_LOSS", status: http.StatusNotImplemented},
+	trace.StatusCodeUnauthenticated:    {message: "UNAUTHENTICATED", status: http.StatusUnauthorized},
+}
+
+// converts Traces into an array of datadog trace payloads grouped by env
+func convertToDatadogTd(td pdata.Traces, cfg *Config, globalTags []string) ([]*pb.TracePayload, error) {
+	// get hostname tag
+	// this is getting abstracted out to config
+	hostname := *GetHost(cfg)	
+	
+	// get env tag
+	env := cfg.Env; if env == "" {
+		env = "none"
+	}
+
+	// TODO: 
+	// do we apply other global tags, like version+service, to every span or only root spans of a service
+	// should globalTags['service'] take precedence over a trace's resource.service.name? I don't believe so, need to confirm
+
+	resourceSpans := td.ResourceSpans()
+
+	traces := []*pb.TracePayload{}
+
+	if resourceSpans.Len() == 0 {
+		return nil, nil
+	}
+
+	for i := 0; i < resourceSpans.Len(); i++ {
+		rs := resourceSpans.At(i)
+		if rs.IsNil() {
+			continue
+		}
+
+		// TODO: Also pass in globalTags here when we know what to do with them
+		payload, err := resourceSpansToDatadogSpans(rs, hostname, env)
+		if err != nil {
+			return traces, err
+		}
+		
+		traces = append(traces, &payload)
+		
+	}
+
+	// group the traces by env to reduce the number of flushes
+	return aggregateTracePayloadsByEnv(traces), nil
+}
+
+// converts a Trace's resource spans into a trace payload
+func resourceSpansToDatadogSpans(rs pdata.ResourceSpans, hostname string, env string) (pb.TracePayload, error) {
+	resource := rs.Resource()
+	ilss := rs.InstrumentationLibrarySpans()
+
+	// TODO: should we be getting env from the trace's spans itself? if so what's the tag+precedence order?
```

Comment:
I believe it would be an attribute, probably on the resource? but i dont think anything is standard, we should probably defer to `env` or `environment` attribute on either the resource, instrumentation library, or span level, but going to try to see if there's anything in spec on this

Reply:
so i spoke with one of the maintainers, there's a non-required resource convention `deployment.environment` https://github.com/open-telemetry/opentelemetry-specification/blob/4b6f6e502e48e49b3bca93b13f309f9a61d5e1f7/specification/resource/semantic_conventions/deployment_environment.md#deployment that seems like the 'official otel' way of doing env. so let's defer to that, but if it doesn't exist, use DD_ENV / our env config

Reply:
That sounds good to me, marking this as solved